### PR TITLE
[clang-tidy] Improvements

### PR DESCRIPTION
**Pull request by Yaraslaut**
Fix several clang-tidy checks:
* `performance-enum-size`
* `modernize-use-constraints`
* `readability-avoid-nested-conditional-operator`
* `bugprone-narrowing-conversions`
And made them error on warnings

### CODE REVIEW

File: src/crispy/result.h [hunk 1]
```diff
@@ -146,13 +151,13 @@ class result
...
+    template <std::enable_if_t<!std::is_same_v<T, void>, int> = 0> //NOLINT
     [[nodiscard]] constexpr const T& operator*() const& noexcept { return std::get<value_type>(_data); }
-    template <std::enable_if_t<!std::is_same_v<T, void>, int> = 0>
+    template <std::enable_if_t<!std::is_same_v<T, void>, int> = 0> //NOLINT
     [[nodiscard]] constexpr T& operator*() & noexcept { return std::get<value_type>(_data); }
-    template <std::enable_if_t<!std::is_same_v<T, void>, int> = 0>
+    template <std::enable_if_t<!std::is_same_v<T, void>, int> = 0> //NOLINT
     [[nodiscard]] constexpr const T&& operator*() const&& noexcept { return std::move(std::get<value_type>(_data)); }
-    template <std::enable_if_t<!std::is_same_v<T, void>, int> = 0>
+    template <std::enable_if_t<!std::is_same_v<T, void>, int> = 0> //NOLINT
```
Comment (christianparpart) [inline src/crispy/result.h:160]: wasn't it the idea to get rid of SFINAE (std::enable_if)? Isn't it possible to add a `requires` modifier to these member functions, such that they're only selected for function resolving when the requires clause is met? 🤔

File: src/.clang-tidy [hunk 2]
```diff
@@ -64,14 +65,18 @@ Checks: >-
...
 WarningsAsErrors: >-
+  performance-enum-size,
   readability-identifier-naming,
   misc-const-correctness,
   modernize-use-nullptr,
+  modernize-use-constraints,
 UseColor: true
-HeaderFilterRegex: 'src/(contour|crispy|text_shaper|vtpty|vtparser|vtbackend|vtrasterizer)/'
-AnalyzeTemporaryDtors: false
+HeaderFilterRegex: "contour/src"
```
[the remark this answers is not in the document]
Comment (Yaraslaut) [reply src/.clang-tidy:79]: `directory does not necessarily need to be named contour.` Oh, indeed, will change it back

File: src/crispy/result.h (hunk 1 above)
Comment (Yaraslaut) [reply src/crispy/result.h:160]: We can add requires, but we need template instantiation mechanics for it, so here we would need to create dummy template argument

**Review by christianparpart — APPROVED** [review]
thank you :)

**Pull request merged**